tests/test_component.py

```diff
@@ -164,7 +166,7 @@ def get_context_data(self, name, css_class="", title="", **attrs):
                     **attrs,
                 }
 
-            def get_template_name(self, context):
+            def get_template_name(self, context):  # noqa
```

Comment:
This was because its method signature conflicts with the signature of Component.get_template_name. The method on the parent class sees 'context' as an optional parameter, whereas the method implementation on the child sees it as required. 

I've fixed this by making 'context' a required parameter of Component.get_template_name. This should have no adverse effects since the method is primarily called by the component object itself and so we can guarantee that context is always passed in.